src/core/lib/security/credentials/external/external_account_credentials.cc

```diff
@@ -197,6 +199,31 @@ RefCountedPtr<ExternalAccountCredentials> ExternalAccountCredentials::Create(
       return nullptr;
     }
   }
+  it = json.object().find("service_account_impersonation");
+  if (it != json.object().end() && it->second.type() == Json::Type::kObject) {
+    options.service_account_impersonation = it->second;
+    auto service_acc_imp_obj_it =
+        options.service_account_impersonation.object().find(
+            "token_lifetime_seconds");
+    if (service_acc_imp_obj_it != it->second.object().end()) {
+      if (service_acc_imp_obj_it->second.type() != Json::Type::kNumber) {
+        *error = GRPC_ERROR_CREATE("token_lifetime_seconds must be a number");
+        return nullptr;
+      }
+      int32_t lifetime =
+          strtol(service_acc_imp_obj_it->second.string().c_str(), nullptr, 10);
+      if (lifetime > IMPERSONATED_CRED_MAX_LIFETIME_IN_SECONDS) {
+        *error = GRPC_ERROR_CREATE(
+            "token_lifetime_seconds must be less than 43200s");
+        return nullptr;
+      }
+      if (lifetime < IMPERSONATED_CRED_MIN_LIFETIME_IN_SECONDS) {
+        *error =
+            GRPC_ERROR_CREATE("token_lifetime_seconds must be more than 600s");
```

Comment:
`600s` and '43200s' are hardcoded in the message - is there a way to externalize it?